### PR TITLE
Using Timestamp from Block for Event Times

### PR DESCRIPTION
**Pull request by petersng**
- Updated to pull the timestamp from the eth blocks to set the timestamp for the events.
    - Doing this to ensure that we can rebuild the state of the data in the contracts precisely every time.  

**TODO**
- Prob need to fix the tests.

### CODE REVIEW

**Comment by innainu** [comment]
The timestamp of the block is the time of inception for the block. So this makes me wonder:
- This timestamp doesn't necessarily match the event timestamp? 
- If there are multiple events within one block, they will all have the same time?

**Comment by innainu** [comment]
Another question about this I don't necessarily know the answer to: 
Since this is the block's inception timestamp, would some of the events that we are listening to have revised timestamps in the event we retrieve the same block bc the block has changed?

**Comment by petersng** [comment]
The reason I made this addition is that when we have to restart and pull the events again, they are tagged all with the same exact date and timestamp.  I think this will be a problem if we need to recover the data from the chain as we won't be able to search via time period otherwise.  

The event will be realized when the block is mined, so the block timestamp made sense to use.  Yes, it's possible that other events will have the same timestamp, but we can also use things like the index number, etc. to ensure ordering.  Also, I can see that events in the same block will likely be related to different listings.  

If you have any other suggestions, let me know, but I don't think the current solution will work.

**Comment by innainu** [comment]
What about the second question though? If a block reorg happens, will the timestamp change when we go back and retrieve the events?

**Comment by innainu** [comment]
Even if multiple events have the same timestamp, if we are going for the definition in the quip doc of "time the event was retrieved" it makes total sense. I'm seeing this timestamp to be most relevant only for DB pulling

**Comment by innainu** [comment]
If we want to search data from the chain, we can just use block number and log index which are more native to the chain anyway.

**Comment by petersng** [comment]
The way it currently works will not work for db pulling.  In our current code, if I start up the system and there a no events, it will grab older events and put them into the DB all at the exact same time.  It will not be possible to actually pull these events out based on dates, because they will be saved with the same date.

**Comment by petersng** [comment]
To answer your second question, not sure, bc it depends on whether we go with the "update the current event" or the "create an update event" approach.

**Comment by petersng** [comment]
Moving the discussion offline.

File: pkg/utils/time.go [hunk 1]
```diff
@@ -30,3 +30,8 @@ func CurrentEpochSecsInInt() int {
 func CurrentEpochNanoSecsInInt64() int64 {
 	return time.Now().UnixNano()
 }
+
+// SecsToNanoSecsInInt64 converts a value from secs to nanoseconds.
+func SecsToNanoSecsInInt64(secs int64) int64 {
```
Comment (innainu) [inline pkg/utils/time.go]: is this fuction actually used anywhere?

Comment (petersng) [reply pkg/utils/time.go]: yes, i see it's missing where the timestamp gets set.  I can add that back.

Comment (innainu) [reply pkg/utils/time.go]: but wait, it doesn't make sense to put it as nano seconds if we are just multiplying by 10^9. we should just adapt to the time unit of the block timestamp

Comment (petersng) [reply pkg/utils/time.go]: The block timestamp is just in secs from epoch.  I just adapted it here bc we are storing as nanoseconds throughout.

Comment (petersng) [reply pkg/utils/time.go]: as per discussion in slack, will change these timestamps to secs from epoch

**Comment by petersng** [comment]
@innainu updated with timestamps in secs

**Pull request merged**